src/components/organism/ToDoList/ToDoList.tsx

```diff
@@ -1,32 +1,70 @@
-import React,{useState} from 'react'
+import React,{useEffect, useState} from 'react'
 import { completetodo, completetodos, deletetodo, deletetodos, selectAlltodos, todoprops} from '../../../redux/features/ToDos'
 import DeleteSelectedToDoDrawer from '../../molecule/DeleteSelectToDoDrawer'
 import DeleteToDoDrawer from '../../molecule/DeleteToDoDrawer'
 import ToDoItem from '../../molecule/ToDoItem'
-import { ToDoListContainer, ToDoListList } from './elements'
+import { Container, NoToDoContainer, NoToDoImg, NoToDop,ToDoListContainer, ToDoListList } from './elements'
 import { ToDoListProps } from './types'
 import { useSelector,useDispatch } from "react-redux"
 import Toast from '../../atom/Toast'
+import Vector from '../../../assets/Vector.png'
+import TextLink from '../../atom/TextLink'
 
-const ToDoList = ({selectionMode,UpdateCallback}: ToDoListProps) => {
-    //global state >
-    const todos = useSelector(selectAlltodos)
+
+const ToDoList = ({selectionMode,UpdateCallback,searchedTodos,addToDo,search}: ToDoListProps) => {
+    //global state>
+    const reduxToDos = useSelector(selectAlltodos)
     const dispatch = useDispatch()
     //<
-    //local states >
+    //compoment state>
     const [selectedToDos,setSelectedToDos] = useState<todoprops[]>([]);
     const [targetToDo,setTargetToDo] = useState<todoprops>({} as todoprops);
     const [deletedrawer,setDeletedrawer] = useState<boolean>(true);
+    let sortedToDos:todoprops[] = []
     //<
-    //sorts between complete and uncomplete >
-    const uncompletedtodos = todos.filter(currentTodo =>{return !currentTodo.complete})
-    const completedtodos = todos.filter(currentTodo =>{return currentTodo.complete})
+    //override/sort>
+        if(search){
+            if(searchedTodos===undefined){
+            }else {
+                if(searchedTodos.length > 0){
+                    sortedToDos = [...searchedTodos];
+                    }
+            }
+        }
+        else {
+            //sorts between complete and uncomplete >
+            const uncompletedtodos = reduxToDos.filter(currentTodo =>{return !currentTodo.complete})
+            const completedtodos = reduxToDos.filter(currentTodo =>{return currentTodo.complete})
+            sortedToDos = [...uncompletedtodos,...completedtodos]
+            //<
+        }
+    //<
+    //>
+    const Filled = sortedToDos.length;
```

Comment:
Also to add, I think it would be better if the variable name corresponds to what it actually contains. In this scenario since the value is the length of `sortedToDos`, it would be better to call it `todosLength`. If what you really need is only to know if it's filled or not, then you can have something like `const isFilled = !!sortedToDos.length;`. This will convert values into boolean.

Reply:
okay, I'll keep it at standard from now on.